torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -38,16 +40,36 @@ const DistAutogradContext& DistAutogradContainer::newContext() {
   }
 
   std::lock_guard<std::mutex> guard(autograd_context_lock_);
-  if (current_context_id_ == std::numeric_limits<int64_t>::max() ||
-      current_context_id_ >
+  if (next_context_id_ == std::numeric_limits<int64_t>::max() ||
+      next_context_id_ >
           (kMaxContextId |
            (static_cast<int64_t>(worker_id_) << kContextIdBits))) {
     throw std::runtime_error("We have run out of autograd context ids!!!");
   }
 
   autograd_context_.emplace(
-      current_context_id_, DistAutogradContext(current_context_id_));
-  return autograd_context_.at(current_context_id_++);
+      std::piecewise_construct,
+      std::forward_as_tuple(next_context_id_),
+      std::forward_as_tuple(next_context_id_));
+
+  current_context_id_ = next_context_id_;
+  return autograd_context_.at(next_context_id_++);
+}
+
+bool DistAutogradContainer::hasValidContext() const {
+  return current_context_id_ != -1;
+}
+
+DistAutogradContext& DistAutogradContainer::currentContext() {
+  TORCH_CHECK(
+      hasValidContext(),
+      "Current thread doesn't have a valid autograd context.");
+  std::lock_guard<std::mutex> guard(autograd_context_lock_);
+  if (autograd_context_.find(current_context_id_) == autograd_context_.end()) {
```

Comment:
Nit: `find()` returns an iterator, which you can use to retrieve the element in question. The way you've written it here, you have to do *two* map lookups in the successful case when only one is necessary.